### PR TITLE
"Project script added "

### PR DESCRIPTION
**Pull request by MallanagoudaB**

### CODE REVIEW

File: sunbird/NEW.xlsx
Comment (adithyadinesh0412) [inline sunbird/NEW.xlsx:1]: Don't Push excel files to git

File: sunbird/ProgramTemplate.xlsx
Comment (adithyadinesh0412) [inline sunbird/ProgramTemplate.xlsx:1]: Don't Push excel files to the repo

File: sunbird/common_config/config.ini [hunk 1]
```diff
@@ -1,39 +1,46 @@
 [pre-prod]
-host = # Add host Ip address followed by a slash / 
-internal_kong_ip = # Add Internal kong Ip address followed by a slash / 
-keyclockapibody = 
+host = http://localhost:3569/
```
Comment (adithyadinesh0412) [inline sunbird/common_config/config.ini:2]: don't push the actual config file to git

Comment (adithyadinesh0412) [reply sunbird/common_config/config.ini:2]: this should just be a sample file. refer existing sample config files

File: sunbird/common_config/config.ini [hunk 2]
```diff
@@ -1,39 +1,46 @@
...
+internal_kong_ip = http://localhost:3569/
+keyclockapibody = "email": "rahul@yahoo.com","password": "password"
```
Comment (adithyadinesh0412) [inline sunbird/common_config/config.ini:4]: How can u expose email and password in git? 
even though its local credentials , its not a recommended practice

File: sunbird/common_config/config.ini [hunk 3]
```diff
@@ -1,39 +1,46 @@
...
+  
 authorization = Bearer # ADD_YOUR_TOKEN_HERE
 authorizationforhost = Bearer # ADD_YOUR_TOKEN_HERE
 appname = diksha
 x-channel-id = 0125747659358699520
-internal-access-token = 183c219f984050c0e89d
+internal-access-token = 8c3a94f0931e01a4940a
```
Comment (adithyadinesh0412) [inline sunbird/common_config/config.ini:10]: this is a confidential data of service. please don't expose.

File: sunbird/common_config/config.ini [hunk 4]
```diff
@@ -1,39 +1,46 @@
 [pre-prod]
-host = # Add host Ip address followed by a slash / 
-internal_kong_ip = # Add Internal kong Ip address followed by a slash / 
-keyclockapibody = 
+host = http://localhost:3569/
+internal_kong_ip = http://localhost:3569/
```
Comment (adithyadinesh0412) [inline sunbird/common_config/config.ini:3]: since both are using interface service url , delete this and keep only one. make relevant change in the script

File: sunbird/common_config/config.ini [hunk 5]
```diff
@@ -49,24 +56,75 @@ editsvgtemp = private/mlcore/api/v1/certificateTemplates/createSvg?baseTemplateI
...
+; programsupdate = /v1/programs/update/{programId}
+; programsdetails = /v1/programs/details/{programId}
+; programsaddRolesInScope = /v1/programs/addRolesInScope/{programId}
+; programsaddEntitiesInScope = /v1/programs/addEntitiesInScope/{programId}
+; programsremoveRolesInScope = /v1/programs/removeRolesInScope/{programId}
+; programsremoveEntitiesInScope = /v1/programs/removeEntitiesInScope/{programId}
+; programslist = /v1/programs/list
+; programsjoin = /v1/programs/join/{programId}
+
+
```
Comment (adithyadinesh0412) [inline sunbird/common_config/config.ini:94]: Remove unwanted code comments

File: sunbird/sample1.py [hunk 6]
```diff
@@ -0,0 +1,5395 @@
...
+ccRootOrgName = None
+ccRootOrgId  = None
+certificatetemplateid = None
+question_sequence_arr = []
+
+# function to map course to program
+
+
+# As per the discussion with products team, course is currently taken down from the product until further notice 
+def courseMapToProgram(accessToken, courseLink, parentFolder):
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:121]: As these functions are not used for the current implementation remove this function and the fn calls as well

File: sunbird/sample1.py [hunk 7]
```diff
@@ -0,0 +1,5395 @@
...
+#                                        courseSolutionId, accessToken)
+#             return courseSolutionId
+#         else:
+#             print("XXXXXXXXX ---- Course to program mapping Failed. ----- XXXXXXXX")
+#     else:
+#         print("XXXXXXXXX --- Course read API failed ----- XXXXXXXXX")
+#         print("XXXXXXXXX --- Check API hit logs ----- XXXXXXXXX")
+
+
+def checkIfObsMappedToProgram(accessToken, obsExt, parentFolder):
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:198]: Remove this fn as its not used in the existing implementation.

File: sunbird/sample1.py [hunk 8]
```diff
@@ -0,0 +1,5395 @@
...
+      "isDeleted" : False,
+      "resourceType" : [ 
+          "program"
+      ],
+      "language" : [ 
+          "English"
+      ],
+      "keywords" : [],
+      "concepts" : [],
+      "userId":"",
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:261]: Add creator userId here

File: sunbird/sample1.py [hunk 9]
```diff
@@ -0,0 +1,5395 @@
...
+      "imageCompression" : {
+          "quality" : 10
+      },
+      "components" : [ 
+      ],
+      "scope" : {
+          "entityType" : "state",
+          "entities" : ["bc75cc99-9205-463e-a722-5326857838f8","8ac1efe9-0415-4313-89ef-884e1c8eee34"],
+          "roles" : ["HM"]
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:270]: Why is scope hard coded?

File: sunbird/sample1.py [hunk 10]
```diff
@@ -0,0 +1,5395 @@
...
+}
+)
+    messageArr.append("Body : " + str(payload))
+    headers = {'X-authenticated-user-token': accessToken,
+               'internal-access-token': config.get(environment, 'internal-access-token'),
+               'Content-Type': 'application/json',
+               'Authorization':config.get(environment, 'Authorization')}
+    
+    # program creation 
+    print(payload)
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:282]: remove this print statement.

File: sunbird/sample1.py [hunk 11]
```diff
@@ -0,0 +1,5395 @@
...
+    pdpmsheet = MainFilePath+ "/pdpmmapping/"
+    if not os.path.exists(pdpmsheet):
+        os.mkdir(pdpmsheet)
+
+    wbproject = xlrd.open_workbook(program_file, on_demand=True)
+    projectSheetNames = wbproject.sheet_names()
+
+    mappingsheet = wbproject.sheet_by_name('Program Details')
+    keysProject = [mappingsheet.cell(1, col_index_env).value for col_index_env in
+                   range(mappingsheet.ncols)]
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:311]: create a funtion for this. as this code is repeated through out the code.

File: sunbird/sample1.py [hunk 12]
```diff
@@ -0,0 +1,5395 @@
...
+
+
+                    if districtentitiesPGM:
+                        entitiesPGM = districtentitiesPGM
+                        EntityType = "district"
+                    else:
+                        entitiesPGM = entitiesPGM
+                        EntityType = "state"
+
+                    scopeEntityType = EntityType
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:498]: Insead of using a 3rd variable `EntityType` directly assign `scopeEntityType`

File: sunbird/sample1.py [hunk 13]
```diff
@@ -0,0 +1,5395 @@
...
+
+                    global entitiesPGMID
+                    # entitiesPGMID = fetchEntityId(parentFolder, accessToken,
+                                                #   entitiesPGM.lstrip().rstrip().split(","), scopeEntityType)
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:502]: why is this commented?

**Review by adithyadinesh0412 — COMMENTED** [review]
I cannot approve this PR to get merged into `release-6.0` branch as the mentioned branch is being used by multiple teams. Please raise this PR to `elevate-project`.

Also Change the name of the main file from sample1.py . Give some meaning full name

File: sunbird/sample1.py [hunk 14]
```diff
@@ -0,0 +1,5395 @@
...
+def valid_file(param):
+    base, ext = os.path.splitext(param)
+    if ext.lower() not in ('.xlsx'):
+        raise argparse.ArgumentTypeError('File must have a csv extension')
+    return param
+
+# function to check environment 
+def envCheck():
+    try:
+        config.get(environment, 'keyclockAPIUrl')
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:661]: change `keyclockAPIUrl` to some other value.

File: sunbird/common_config/config.ini [hunk 15]
```diff
@@ -49,24 +56,75 @@ editsvgtemp = private/mlcore/api/v1/certificateTemplates/createSvg?baseTemplateI
...
+; keyclockapibody = '{"email": "rahul@yahoo.com","password": "password"}'
+; email = rahul@yahoo.com
+; password = password
+; internal-access-token = 8c3a94f0931e01a4940a
+; keyclockapiurl = user/v1/account/login
+
+
+; DEV_testing
+host = https://dev.elevate-apis.shikshalokam.org/
+host1 = https://dev.elevate-apis.shikshalokam.org/
```
Comment (adithyadinesh0412) [inline sunbird/common_config/config.ini:112]: what is host1? Can we not do it using a single variable?

File: sunbird/sample1.py [hunk 16]
```diff
@@ -0,0 +1,5395 @@
...
+    orgIds = []
+    organisations = str(OrgName).split(",")
+    for org in organisations:
+        orgBody = {"id": "",
+                   "ts": "",
+                   "params": {
+                       "msgid": "",
+                       "resmsgid": "",
+                       "status": "success"
+                   },
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:872]: check if this body is sending only relavant values.

File: sunbird/sample1.py [hunk 17]
```diff
@@ -0,0 +1,5395 @@
...
+                   "request": {
+                       "filters": {
+                           "orgName": str(org).strip()
+                       }
+                   }}
+
+        responseOrgSearch = requests.request("POST", url, headers=headers, data=json.dumps(orgBody))
+        print(responseOrgSearch)
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:880]: remove this print. 
All debug print statements should be removed from the code.

File: sunbird/sample1.py [hunk 18]
```diff
@@ -0,0 +1,5395 @@
...
+
+
+def fetchEntityId(solutionName_for_folder_path, accessToken, entitiesNameList, scopeEntityType):
+    urlFetchEntityListApi = config.get(environment, 'host1')+config.get(environment, 'searchForLocation')
+    headerFetchEntityListApi = {
+        'Content-Type': config.get(environment, 'Content-Type'),
+        'X-authenticated-user-token': accessToken,
+        'internal_access_token': config.get(environment, 'internal-access-token'),
+    }
+    payload = {
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:922]: is this payload related to entity search?

File: sunbird/sample1.py [hunk 19]
```diff
@@ -0,0 +1,5395 @@
...
+      "bc75cc99-9205-463e-a722-5326857838f8",
+      "8ac1efe9-0415-4313-89ef-884e1c8eee34"
+    ],
+    "roles": [
+      "HM"
+    ]
+  },
+  "requestForPIIConsent": true
+}
+    responseFetchEntityListApi = requests.post(url=urlFetchEntityListApi, headers=headerFetchEntityListApi,data=json.dumps(payload))
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:954]: recheck the payload we are passing

File: sunbird/sample1.py [hunk 20]
```diff
@@ -0,0 +1,5395 @@
...
+
+           if str(dictTasksDetails["Mandatory task(Yes or No)"]).strip().strip().lower() == "no":
+               isDeletable = "TRUE"
+           else:
+               isDeletable = "FALSE"
+           task_values = [taskName, taskId, taskDescription, taskType, hasAParentTask, parentTaskOperator, parentTaskValue,
+                          parentTaskId, taskSolutionType, solutionSubType, solutionId, isDeletable]
+           task_lr_value_count = 1
+           for task_lr in range(0, int(taskLearningResource_count)):
+               task_lr_name = str(dictTasksDetails["learningResources" + str(task_lr_value_count)
```
Comment (adithyadinesh0412) [inline sunbird/sample1.py:5317]: now we have 2 different approaches. 
1 with program file
2 with project file.

SO i reccommend instead of having the command line argument with same name tag, use different for different approaches. 
ex : --programFile and --resourceFile
Identify the approach from the command line argument passed.

Comment (adithyadinesh0412) [inline sunbird/sample1.py:5333]: remove these prints

Comment (adithyadinesh0412) [inline sunbird/sample1.py:5000]: remove

File: sunbird/common_config/config.ini (hunk 15 above)
Comment (MallanagoudaB) [reply sunbird/common_config/config.ini:112]: yes @adithyadinesh0412  i though you will test and check in your local so i kept all the data in congig.ini
